src/s3.rs

```diff
@@ -6699,14 +6699,10 @@ impl PrefixParser {
     fn parse_xml<T: Peek + Next>(tag_name: &str, stack: &mut T) -> Result<Prefix, XmlParseError> {
         try!(start_element(tag_name, stack));
         // this could be an empty, self-closing tag:
-        match end_element(tag_name, stack) {
-            Ok(()) => Ok("".to_string()),
-            Err(_) => {
-                let obj = try!(characters(stack));
-                try!(end_element(tag_name, stack));
-                Ok(obj)
-            }
-        }
+        // If it is, characters() returns an empty string already, so why bother ?
```

Comment:
Let's combine this with the comment above to something like this:

`Read characters: if it's an empty, self closing tag we'll get an empty string back and move on.`